src/dispatchers.ts

```diff
@@ -300,45 +306,66 @@ export async function inboxErrorHandler(
     logging.error('Error handling incoming activity', { error });
 }
 
-function convertJsonLdToRecipient(result: any): Recipient {
-    return {
-        ...result,
-        id: new URL(result.id),
-        inboxId: new URL(result.inbox),
-        endpoints: result.endpoints?.sharedInbox != null
-            ? { sharedInbox: new URL(result.endpoints.sharedInbox) }
-            : null,
-    };
-}
-
 export async function followersDispatcher(
     ctx: Context<ContextData>,
     handle: string,
+    cursor: string | null,
 ) {
     logging.info('Followers Dispatcher');
+
+    const offset = parseInt(cursor ?? '0');
+    let nextCursor: string | null = null;
+
     let items: Recipient[] = [];
+
     const fullResults = (await ctx.data.db.get<any[]>(['followers', 'expanded']) ?? [])
         .filter((v, i, results) => {
             // Remove duplicates
             return results.findIndex((r) => r.id === v.id) === i;
         });
+
     if (fullResults) {
-        items = fullResults.map(convertJsonLdToRecipient)
+        nextCursor = fullResults.length > offset + FOLLOWERS_PAGE_SIZE
+            ? (offset + FOLLOWERS_PAGE_SIZE).toString()
+            : null;
+
+        items = fullResults.slice(offset, offset + FOLLOWERS_PAGE_SIZE);
     } else {
         const results = [
             // Remove duplicates
             ...new Set(
                 (await ctx.data.db.get<string[]>(['followers'])) || []
             )
         ];
-        const actors = items = (await Promise.all(results.map((result) => lookupActor(ctx, result))))
-            .filter((item): item is Actor => isActor(item))
-        const toStore = await Promise.all(actors.map(actor => actor.toJsonLd() as any));
+
+        nextCursor = results.length > offset + FOLLOWERS_PAGE_SIZE
+            ? (offset + FOLLOWERS_PAGE_SIZE).toString()
+            : null;
+
+        const slicedResults = results.slice(offset, offset + FOLLOWERS_PAGE_SIZE);
+
+        const actors = (
+            await Promise.all(
+                slicedResults.map((result) => lookupActor(ctx, result))
+            )
+        ).filter((item): item is Actor => isActor(item))
```

Comment:
We do a filter here _after_ slicing - which means we may get a smaller page than expected, I think it's fine for now - but do you mind adding a TODO comment to the code - we really should be able to remove this check

Reply:
🤔 I think this was to get around some typing issue, maybe there is a better approach

Reply:
Ah, i see what you mean, this was in the original actor resolution block. Will add a note 👍

Reply:
Note added and tracking [here](https://linear.app/ghost/issue/AP-549/following-dispatcher-may-return-less-results-than-defined-page-size)